.github/workflows/required-tests.yml

```diff
@@ -0,0 +1,211 @@
+name: Required Tests
+
+on:
+    push:
+        branches:
+            - master
+            - gh-actions
+    pull_request:
+        branches:
+            - master
+
+jobs:
+    # "Build" job
+    build:
+      # This job tries to build PKI from src on a fresh docker container.
+      # The docker container is spawned by github itself and we merely just
+      # run the build commands. We then upload the artifact for consumption
+      # by the test jobs + for the public to consume. This job **does not**
+      # upload any build logs as they are visible in the log console itself.
+ 
+      name: Build PKI
+      runs-on: ubuntu-latest
+      container: fedora:${{ matrix.os }}
+      strategy:
+          matrix:
+            os: ['30', '31']
+      steps:
+          - name: Update base image
+            run: |
+                  set -x &&
+                  dnf update -y &&
+                  dnf install -y dnf-plugins-core rpm-build git
+
+          - name: Clone the repository
+            uses: actions/checkout@v2
+
+          - name: Install PKI build deps
+            run: |
+                  # dnf copr enable @pki/master
+                  dnf builddep -y --allowerasing --spec ./pki.spec &&
+                  dnf clean all
+
+          - name: Build PKI packages
+            run: ./build.sh --with-timestamp --with-commit-id --work-dir=../packages rpm
+
+          - name: Compress RPMS
+            run: tar -czf pki-rpms.tar.gz ../packages/RPMS/*
+
+          # upload-artifact runs on host-vm rather than inside the container. Fixed in v2 (unreleased)
+          # Bug: https://github.com/actions/upload-artifact/issues/13#issuecomment-532936650
+          - name: Upload RPM artifacts
+            uses: actions/upload-artifact@v1
+            with:
+              name: pki-build-${{ matrix.os }}
+              path: pki-rpms.tar.gz
```

Comment:
Could we upload the individual rpm files instead of the tarball? Right now if you try to download the uploaded file it will return a zip file containing the tarball, so the tarball is redundant and adding an extra step to unpack the content. Or if there's a way to tell GH to avoid creating the zip file that will work too.

Reply:
@edewata Currently, `upload-artifact@v1` zips the files and uploads it. So, you can't prevent this.

OTOH, the `v1` can't handle dirs or wildcard pattern. This is available in[ `v2-preview`](https://github.com/actions/upload-artifact/issues/62#issue-580668111). I ignored v2 as there was a clear warning that things could break overtime. Since we all are new to GH actions, I prioritized stabilizing the pipeline before using new features :smile:

Reply:
Sure, we can migrate later to v2 when it's finalized.